lib/seattleflu/id3c/cli/command/etl/redcap.py

```diff
@@ -44,3 +69,628 @@ def normalize_net_id(net_id: str=None) -> Optional[str]:
+
+    def residence_census_tract(db: DatabaseSession, lat_lng: Tuple[float, float],
+            housing_type: str, system_identifier: str) -> Optional[dict]:
+        """
+        Creates a new Location Resource for the census tract containing the given
+        *lat_lng* coordintes and associates it with the given *housing_type*.
+        """
+        location = location_lookup(db, lat_lng, 'tract')
+
+        if location and location.identifier:
+            return create_location(
+                f"{system_identifier}/location/tract", location.identifier, housing_type
+            )
+        else:
+            LOG.debug("No census tract found for given location.")
+            return None
+
+    def create_location(system: str, value: str, location_type: str, parent: str=None) -> dict:
+        """ Returns a FHIR Location resource. """
+        location_type_system = "http://terminology.hl7.org/CodeSystem/v3-RoleCode"
+        location_type_map = {
+            "residence": "PTRES",
+            "school": "SCHOOL",
+            "work": "WORK",
+            "site": "HUSCS",
+            "lodging": "PTLDG",
+        }
+
+        location_type_cc = create_codeable_concept(location_type_system,
+            location_type_map[location_type])
+        location_identifier = create_identifier(system, value)
+        part_of = None
+        if parent:
+            part_of = create_reference(reference_type="Location", reference=parent)
+
+        return create_location_resource([location_type_cc], [location_identifier], part_of)
+
+    lodging_options = [
+        'shelter',
+        'afl',
+        'snf',
+        'ltc',
+        'be',
+        'pst',
+        'cf',
+        'none',
+    ]
+
+    if housing_type in lodging_options:
+        housing_type = 'lodging'
+    else:
+        housing_type = 'residence'
+
+    address = {
+        'street': primary_street_address,
+        'secondary': secondary_street_address,
+        'city': city,
+        'state': state,
+        'zipcode': zipcode
+    }
+
+    lat, lng, canonicalized_address = get_response_from_cache_or_geocoding(address, cache)
+    if not canonicalized_address:
+        return []  # TODO
+
+    tract_location = residence_census_tract(db, (lat, lng), housing_type, system_identifier)
+    # TODO what if tract_location is null?
+    tract_full_url = generate_full_url_uuid()
+    tract_entry = create_resource_entry(tract_location, tract_full_url)
+
+    address_hash = generate_hash(canonicalized_address)
+    address_location = create_location(
+        f"{system_identifier}/location/address",
+        address_hash,
+        housing_type,
+        tract_full_url
+    )
+    address_entry = create_resource_entry(address_location, generate_full_url_uuid())
+
+    return [tract_entry, address_entry]
+
+
+def create_site_reference(location: str, site_map: dict, default_site: str,
+    system_identifier: str) -> Optional[Dict[str,dict]]:
```

Comment:
Seeing how this is used in the [Childcare ETL](https://github.com/seattleflu/id3c-customizations/pull/154/files#diff-3360a443a510e893404249bad41afeca5eefb61e2d9f7e684e6c9369e2777226R181), the `location` and `site_map` parameters should be optional so the caller doesn't have to pass the `None` values.

Reply:
For sure. I did that in the new branch for the redo.